src/test/java/io/axoniq/axonserver/connector/event/impl/PersistentEventChannelImplTest.java

```diff
@@ -0,0 +1,94 @@
+                                                                                              UUID.randomUUID()
+                                                                                                  .toString()).build();
+        eventChannel = connectionFactory.connect("second").eventChannel();
+        eventChannel.deletePersistentStream("sample-events").get();
+    }
+
+    @Test
+    @Disabled
+    void openPersistedStream() throws ExecutionException, InterruptedException, TimeoutException {
+        PersistentStream streams = eventChannel.openPersistentStream(
+                "sample-events", new PersistedStreamProperties(
+                        null,
+                        2,
+                        "AggregateIdentifier",
+                        null,
+                        0,
+                        null));
+        streams.onSegmentOpened(segmentEventStream -> {
+            Flux.from(new ResultStreamPublisher<>(() -> segmentEventStream))
+                .publishOn(Schedulers.newSingle("segment-" + segmentEventStream.segment()))
+                .subscribe(new Subscriber<EventWithToken>() {
+                    private Subscription subscription;
+
+                    @Override
+                    public void onSubscribe(Subscription subscription) {
+                        this.subscription = subscription;
+                        subscription.request(100);
+                    }
+
+                    @Override
+                    public void onNext(EventWithToken event) {
+                        logger.info("{}: next available -> {} ", segmentEventStream.segment(), event);
+                        segmentEventStream.acknowledge(event.getToken());
+                        subscription.request(1);
+                    }
+
+                    @Override
+                    public void onError(Throwable throwable) {
+                        logger.warn("{}: exception ", segmentEventStream.segment(), throwable);
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        logger.info("{}: closed ", segmentEventStream.segment());
+                    }
+                });
+        });
+
+        eventChannel.appendEvents(Event.newBuilder()
+                                       .setAggregateIdentifier("1234")
+                                       .build()).get(1, TimeUnit.SECONDS);
+        eventChannel.appendEvents(Event.newBuilder()
+                                       .setAggregateIdentifier("1235")
+                                       .build()).get(1, TimeUnit.SECONDS);
+        eventChannel.appendEvents(Event.newBuilder()
+                                       .setAggregateIdentifier("1236")
+                                       .build()).get(1, TimeUnit.SECONDS);
+
+        long last = eventChannel.getLastToken().get();
+        Thread.sleep(2000);
```

Comment:
There is an `assertWithin` in the `AssertUtils` class which would prevent the need for arbitrary sleeps.

There don't seem to be any assertions in this test either.